patch.go

```diff
@@ -147,7 +156,12 @@ func generateEnvironmentVariables(config agentConfig) []corev1.EnvVar {
 	for name, value := range config.Environment {
 		vars = append(vars, corev1.EnvVar{Name: name, Value: value})
 	}
-	return vars
+	for _, ev := range vars {
+		if errs := validation.IsEnvVarName(ev.Name); len(errs) != 0 {
+			return nil, fmt.Errorf("failed to validate variable name: %v", errs)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to validate environment variable %q: %v", errs)
```

The error strings returned by IsEnvVarName don't include the name, AFAICS.

Reply:
Good catch!